src/appshell/qml/DevTools/Gallery/GeneralComponentsGallery.qml

```diff
@@ -389,6 +390,43 @@ Rectangle {
         }
     }
 
+    Component {
+        id: visibleButtonSample
+
+        Column {
+            spacing: 8
+
+            width: 200
+
+            VisibleButton {
+                text: "Visible Button"
+
+                onVisibleEdited: {
```

Comment:
not quite sure why would we need to define it every time, I believe it should happen internally and you could just receive a signal for an additional logic

Reply:
But it is already signal) 
You can see how I use this signal in the inspector

Reply:
I haven't got the point :)

1 - it's still makes sense to keep the notification signal
2 - I'd expect that "VisibilityButton" component would take care about isVisible property by itself

Reply:
We usually bind the isVisible property to the model. When the user tries to change a property, the value is sent to the model. There may be logic in the model that prohibits changing the property, then the model will ignore and nothing will change.

The component should just show the data that was provided to it, and not manage something